From 75e9dcb08b56c64b88c98bd249607d7fefa8469a Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@intel.com>
Date: Fri, 18 Sep 2026 13:43:20 -0700
Subject: [PATCH] tst_Parser: don't hardcode the page size in all architectures

It is fixed to 4096 bytes on x86, but on others it might be different.
For them, ask the runtime what the actual size is.

Fixes #324.

Signed-off-by: Thiago Macieira <thiago.macieira@intel.com>
---
 tests/parser/tst_parser.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/parser/tst_parser.cpp b/tests/parser/tst_parser.cpp
index f937458..93ca26b 100644
--- a/tests/parser/tst_parser.cpp
+++ b/tests/parser/tst_parser.cpp
@@ -157,7 +157,11 @@ struct ParserWrapper
     uint8_t *allocateMemory(size_t);
     void freeMemory();
 
+#if defined(Q_PROCESSOR_X86)
     static const size_t PageSize = 4096;
+#else
+    static inline const size_t PageSize = getpagesize();
+#endif
     static inline size_t mmapAllocation(size_t n)
     {
         // round up and add one page
-- 
2.55.0

